File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int i;
        for(i=0;i<nums.size();i++)
        {
            if(nums[i]==target)
            {
                break;
            }
            else if(nums[i]>target)
            {
                break;
            }
        }
        return i;
        }
        
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int low=0,high=nums.length-1;
        int max=Integer.MIN_VALUE;
        int maxIdx=-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(nums[mid]==target){
                return mid;
            }
            else if(nums[mid]<target){
                if(nums[mid]>max){
                    max=nums[mid];
                    maxIdx=mid;
                }
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        return maxIdx+1;
    }
}